pool/pubsub.go

```diff
@@ -0,0 +1,86 @@
+package pool
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+
+	"github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/codec/dagcbor"
+	"github.com/ipld/go-ipld-prime/node/bindnode"
+	"github.com/ipld/go-ipld-prime/schema"
+	"github.com/multiformats/go-multiaddr"
+)
+
+const Version0 = "0"
+
+var (
+	PubSubPrototypes struct {
+		Announcement schema.TypedPrototype
+	}
+
+	//go:embed pubsub.ipldsch
+	schemaBytes []byte
+)
+
+type (
+	AnnouncementType int
+	Announcement     struct {
```

Comment:
Couldn't we use the structure in "[pool/pubsub.ipldsch](https://github.com/functionland/go-fula/pull/72/files#diff-51033d30d95e3c4f6a93f1eabb2bb922a8459fa16afed568bc4e9c46026c2444)" instead of redefining here or it is something different here? (Sorry this might be lack of my golang experience)